lib/WebDriverCheckbox.php

```diff
@@ -0,0 +1,290 @@
+        }
+
+        $this->byVisibleText($text, false, false);
+    }
+
+    public function deselectByVisiblePartialText($text)
+    {
+        if (!$this->isMultiple()) {
+            throw new UnsupportedOperationException('You may only deselect checkboxes');
+        }
+
+        $this->byVisibleText($text, true, false);
+    }
+
+    /**
+     * Selects or deselects a checkbox or a radio button by its value.
+     *
+     * @param string $value
+     * @param bool $select
+     * @throws NoSuchElementException
+     */
+    private function byValue($value, $select = true)
+    {
+        $matched = false;
+        foreach ($this->getRelatedElements($value) as $element) {
+            $select ? $this->selectOption($element) : $this->deselectOption($element);
+            if (!$this->isMultiple()) {
+                return;
+            }
+
+            $matched = true;
+        }
+
+        if (!$matched) {
+            throw new NoSuchElementException(
+                sprintf('Cannot locate option with value: %s', $value)
+            );
+        }
+    }
+
+    /**
+     * Selects or deselects a checkbox or a radio button by its index.
+     *
+     * @param int $index
+     * @param bool $select
+     * @throws NoSuchElementException
+     */
+    private function byIndex($index, $select = true)
+    {
+        $options = $this->getRelatedElements();
+        if (!isset($options[$index])) {
+            throw new NoSuchElementException(sprintf('Cannot locate option with index: %d', $index));
+        }
+
+        $select ? $this->selectOption($options[$index]) : $this->deselectOption($options[$index]);
+    }
+
+    /**
+     * Selects or deselects a checkbox or a radio button by its visible text.
+     *
+     * @param string $text
+     * @param bool $partial
+     * @param bool $select
+     */
+    private function byVisibleText($text, $partial = false, $select = true)
+    {
+        foreach ($this->getRelatedElements() as $element) {
+            $normalizeFilter = sprintf($partial ? 'contains(normalize-space(.), %s)' : 'normalize-space(.) = %s', XPathEscaper::escapeQuotes($text));
+
+            $xpath = 'ancestor::label';
+            $xpathNormalize = sprintf('%s[%s]', $xpath, $normalizeFilter);
+            if (null !== $id = $element->getAttribute('id')) {
+                $idFilter = sprintf('@for = %s', XPathEscaper::escapeQuotes($id));
+
+                $xpath .= sprintf(' | //label[%s]', $idFilter);
+                $xpathNormalize .= sprintf(' | //label[%s and %s]', $idFilter, $normalizeFilter);
+            }
+
+            try {
+                $element->findElement(WebDriverBy::xpath($xpathNormalize));
+            } catch (NoSuchElementException $e) {
+                if ($partial) {
+                    continue;
+                }
+
+                try {
+                    // Since the mechanism of getting the text in xpath is not the same as
```

Comment:
I'm not sure that this case really occurs. But as the `WebDriverSelect` already does that, I've kept this behavior.